core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeSorterSpillReader.java

```diff
@@ -73,7 +73,9 @@ public UnsafeSorterSpillReader(
     }
 
     final InputStream bs =
-        new NioBufferedFileInputStream(file, (int) bufferSizeBytes);
+        new ReadAheadInputStream(
+            new NioBufferedFileInputStream(file, (int) bufferSizeBytes),
+            (int)bufferSizeBytes, (int)bufferSizeBytes / 2);
```

Comment:
Our test shows best result was obtained when threshold = 0.5* buffer size. I made it configurable anyways in case someone wants to play with it.